File: src/main/java/com/lessmarkup/userinterface/model/user/ChangePasswordModel.java
package com.lessmarkup.userinterface.model.user;

import com.lessmarkup.TextIds;
import com.lessmarkup.interfaces.recordmodel.InputField;
import com.lessmarkup.interfaces.recordmodel.InputFieldType;
import com.lessmarkup.interfaces.recordmodel.RecordModel;

public class ChangePasswordModel extends RecordModel<ChangePasswordModel> {

    private String password;

    public ChangePasswordModel() {
        super(TextIds.CHANGE_PASSWORD);
    }

    @InputField(type = InputFieldType.PASSWORD_REPEAT, textId = TextIds.PASSWORD)
    public void setPassword(String password) { this.password = password; }
    public String getPassword() { return password; }
}

File: src/main/scala/com/lessmarkup/userinterface/model/user/ChangePasswordModel.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.userinterface.model.user

import com.lessmarkup.TextIds
import com.lessmarkup.interfaces.annotations.{InputField, InputFieldType}
import com.lessmarkup.interfaces.recordmodel.RecordModel

class ChangePasswordModel extends RecordModel[ChangePasswordModel](TextIds.CHANGE_PASSWORD) {
  private var password: String = null

  @InputField(fieldType = InputFieldType.PASSWORD_REPEAT, textId = TextIds.PASSWORD) def setPassword(password: String) {
    this.password = password
  }

  def getPassword: String = {
    password
  }
}